src/components/Header/Header.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+type Props = {
+  todos: Todo[];
+  inputRef: React.MutableRefObject<HTMLInputElement | null>;
+  onAdd: (title: string) => Promise<void>;
+  errorMessage: ErrorMessage;
+  setErrorMessage: (error: ErrorMessage) => void;
+  isInputDisabled: boolean;
+  isTodoLoading: boolean;
+};
+
+export const Header: React.FC<Props> = props => {
+  const {
+    todos,
+    onAdd,
+    errorMessage,
+    setErrorMessage,
+    inputRef,
+    isInputDisabled,
+    isTodoLoading,
+  } = props;
+
+  const allTodoCompleted = todos.every(todo => todo.completed);
+  const [title, setTitle] = useState('');
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputRef, isInputDisabled, isTodoLoading]);
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
```

Comment:
It is important to stick to one way of creating functions
```suggestion
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```